### PR TITLE
chore(docs): reflect changes to use env variables in doker-compose

### PR DESCRIPTION
**Pull request by daniel-herrero**
gif

This PR updates the docker installation guide to reflect changes in this other [PR](https://github.com/kaleidos-ventures/taiga-docker/pull/104) 
Most important change: 
> Centralize most common env vars from docker-compose.yml to an .env file to abstract complexity

### CODE REVIEW

File: setup-production.adoc [hunk 1]
```diff
@@ -31,217 +31,172 @@ $ cd taiga-docker/
...
 
-This configuration is likely to suit what you need. Edit environment variables in `docker-compose.yml` and `docker-compose-inits.yml`. Have in mind that some of the variables are in both files, and you need to edit both.
+We've exposed the ** Basic configuration ** and the most commonly modified settings in Taiga to an `.env` file. We strongly recommend you to change it, or at least review its content, to avoid using the default values.
 
-**Configuration** variables are in `docker-compose.yml` with default values that we strongly recommend that you change or review. Those variables are needed to run Taiga. Apart from this configuration, you can have some **customisation** in Taiga, that add features which are disabled by default. Find those variables in **Customisation** section and add the corresponding environment variables whenever you want to enable them.
+Both `docker-compose.yml` and `docker-compose-inits.yml` will read from this file to populate their environment variables, so, initially you don't need to change them. Edit those files just in case you require to enable **Additional customization**, or an **Advanced configuration**.
```
Comment (yamila-moreno) [inline setup-production.adoc]: these files

File: setup-production.adoc [hunk 2]
```diff
@@ -31,217 +31,172 @@ $ cd taiga-docker/
...
 [source, bash]
 ----
-TAIGA_SECRET_KEY: "taiga-back-secret-key"
-TAIGA_SITES_SCHEME: "https"
-TAIGA_SITES_DOMAIN: "taiga.mycompany.com"
-TAIGA_SUBPATH: ""
+TAIGA_SCHEME=https
+TAIGA_DOMAIN=taiga.mycompany.com
+SUBPATH=""
+WEBSOCKETS_SCHEME=wss://taiga.mycompany.com
```
Comment (yamila-moreno) [inline setup-production.adoc]: this should have only the scheme

File: setup-production.adoc [hunk 3]
```diff
@@ -31,217 +31,172 @@ $ cd taiga-docker/
...
 [source, bash]
 ----
-TAIGA_SECRET_KEY: "taiga-back-secret-key"
-TAIGA_SITES_SCHEME: "https"
-TAIGA_SITES_DOMAIN: "mycompany.com"
-TAIGA_SUBPATH: "/taiga"
+TAIGA_SCHEME=https
+TAIGA_DOMAIN=mycompany.com
+SUBPATH="/taiga"
+WEBSOCKETS_SCHEME=wss://mycompany.com
```
Comment (yamila-moreno) [inline setup-production.adoc]: same, should only have the schem

File: setup-production.adoc [hunk 4]
```diff
@@ -31,217 +31,172 @@ $ cd taiga-docker/
...
-TAIGA_SUBPATH: "/taiga"
-----
+.**Email settings**
 
-**Service: taiga-events**
-[source, bash]
-----
-TAIGA_SECRET_KEY: "taiga-back-secret-key"
-----
+By default, emails will be printed in the standard output (`EMAIL_BACKEND=**console**`). If you have a SMTP service, change it to `EMAIL_BACKEND=**smtp**` and configure the rest of these variables:
```
Comment (yamila-moreno) [inline setup-production.adoc]: As here we tipically find many questions, I'd add something like. "Your SMTP provider will give you the proper values for these settings."

File: setup-production.adoc [hunk 5]
```diff
@@ -31,217 +31,172 @@ $ cd taiga-docker/
...
+RABBITMQ_ERLANG_COOKIE=secret-erlang-cookie  # unique value shared by any connected instance of RabbitMQ
 ----
 
-More info about those variables can be found link:https://docs.djangoproject.com/en/3.1/ref/settings/#csrf-cookie-secure[here].
 
-.Email Settings
+.**Attachments settings**
 
-By default, email is configured with the *console* backend, which means that the emails will be shown in the stdout. If you have an smtp service, uncomment the "Email settings" section in `docker-compose.yml` and configure those environment variables:
+You can configure how long the attachments will be accessible by changing the token expiration timer. After that amount of seconds the token will expire.
```
Comment (yamila-moreno) [inline setup-production.adoc]: It seems to me that we are saying that the attachment will no longer be available after the expiration-age, and, if i recall properly it's more something like:

"the token will expire, but you can always get a new attachment url with an active token".

File: setup-production.adoc [hunk 6]
```diff
@@ -31,217 +31,172 @@ $ cd taiga-docker/
...
-**Service: taiga-back**
-[source, bash]
-----
-RABBITMQ_USER: taiga
-RABBITMQ_PASS: taiga
-----
+All these customization options are by default disabled and require you to edit `docker-compose.yml` and `docker-compose-inits.yml`.
 
-Two other variables `EVENTS_PUSH_BACKEND_URL` and `CELERY_BROKER_URL` can also be used to set the events push backend URL and celery broker URL.
+You should add the corresponding environment variables in the proper services with a valid value in order to enable them. Have in mind that some variables are in both files, and you need to edit both.
```
Comment (yamila-moreno) [inline setup-production.adoc]: which customizacion variables are used in both files?

File: setup-production.adoc [hunk 7]
```diff
@@ -31,217 +31,172 @@ $ cd taiga-docker/
...
 
-The attachments will be accesible with a token during MAX_AGE (in seconds). After that, the token will expire.
-
-=== Customisation
+More info about those variables can be found link:https://docs.djangoproject.com/en/3.1/ref/settings/#csrf-cookie-secure[here].
 
-All these features are disabled by default. You should add the corresponding environment variables with a proper value to enable them.
 
-.Registration Settings
+.**Events and asynchronous tasks**
```
Comment (yamila-moreno) [inline setup-production.adoc]: I'd delete this section.
It's very specific; doesn't add a new feature and it's more confusing than useful.

File: setup-production.adoc [hunk 8]
```diff
@@ -31,217 +31,172 @@ $ cd taiga-docker/
...
 [source, bash]
 ----
-PUBLIC_REGISTER_ENABLED: "true"
+EVENTS_PUSH_BACKEND_URL: "amqp://taiga:taiga@taiga-events-rabbitmq:5672/taiga"
+CELERY_BROKER_URL: "amqp://taiga:taiga@taiga-async-rabbitmq:5672/taiga"
 ----
 
-If you want to allow a public register, configure this variable to "True". By default is "False". Should be the same as this var in `taiga-front` and `taiga-back`.
-
 **Important**: Taiga (in its default configuration) disables both Gitlab or Github oauth buttons whenever the public registration option hasn't been activated. To be able to use Github/ Gitlab login/registration, make sure you have public registration activated on your Taiga instance.
```
Comment (yamila-moreno) [inline setup-production.adoc:188]: this "important" was related to "public registration" (before the changes) and now is kind of hanging nowhere.

File: setup-production.adoc [hunk 9]
```diff
@@ -340,33 +295,35 @@ JIRA_IMPORTER_CERT: "cert-from-jira"
...
+Service: `taiga-front`
 [source, bash]
 ----
 ENABLE_TRELLO_IMPORTER: "true"
 ----
 
-=== Advanced configuration and customisation
+=== Advanced configuration
+
+The advanced configuration **will ignore** the environment variables in `docker-compose.yml` or `docker-compose-inits.yml`.
```
Comment (yamila-moreno) [inline setup-production.adoc]: I'd add: "Skip this section if you're using env vars."

**Comment by yamila-moreno** [comment]
This changes made the documentation sooo better !!! KUDOS!

File: setup-production.adoc (hunk 2 above)
Comment (daniel-herrero) [reply setup-production.adoc]: True, well caught. Now it's easier for the user. thanks

File: setup-production.adoc (hunk 4 above)
Comment (daniel-herrero) [reply setup-production.adoc]: I'll leave it this way: "If you have your own SMTP service, change it to `EMAIL_BACKEND=**smtp**` and configure the rest of these variables with the proper values supplied by your SMTP provider"

File: setup-production.adoc (hunk 5 above)
Comment (daniel-herrero) [reply setup-production.adoc]: I find it perfect. I'll add your comment.

File: setup-production.adoc (hunk 6 above)
Comment (daniel-herrero) [reply setup-production.adoc]: It applies to any environment variable changed in the `"Additional customization".

File: setup-production.adoc (hunk 7 above)
Comment (daniel-herrero) [reply setup-production.adoc]: I agree. I'll remove the chapter

File: setup-production.adoc (hunk 8 above)
Comment (daniel-herrero) [reply setup-production.adoc:188]: Mea culpa. I'll change its order to be at the end of "public registration". I moved here the Django admin section not being aware of this part. Thanks

File: setup-production.adoc (hunk 9 above)
Comment (daniel-herrero) [reply setup-production.adoc]: Nice, I'll add it.

File: upgrades-docker-migrate.adoc [hunk 10]
```diff
@@ -0,0 +1,57 @@
+= Migrate to new docker settings
+:toc: left
+:toclevels: 1
+:numbered:
+:source-highlighter: pygments
+:pygments-style: friendly
+
+While `taiga-docker` has been a pretty successful way for many people to have a Taiga self-hosted, the Taiga team receives a bunch of questions about the docker configuration. We have released a new `taiga-docker` version with the main settings based in an `.env` file. The main goal behind this change is to make it way easier to have a basi Taiga up and running.
```
Comment (daniel-herrero) [inline upgrades-docker-migrate.adoc]: I think there's a typo. Did you mean to say `basic Taiga` ?

File: upgrades-docker-migrate.adoc [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+
+[[taigadckr-to-taigadckr]]
+== Summary
+
+If you are familiar with docker and docker compose, this is the sensible summary:
+- save the modifications made in the `docker-compose.yml` and `docker-compose-inits.yml` files
+- update the repository `taiga-docker`
+- add the configuration values to the `.env` file
+- add the customisation values to the `docker-compose.yml` file
+- launch all services again
```
Comment (daniel-herrero) [inline upgrades-docker-migrate.adoc:21]: The list isn't being rendered correctly. You missed a new line before the list itself.

image

File: upgrades-docker-migrate.adoc [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+
+While `taiga-docker` has been a pretty successful way for many people to have a Taiga self-hosted, the Taiga team receives a bunch of questions about the docker configuration. We have released a new `taiga-docker` version with the main settings based in an `.env` file. The main goal behind this change is to make it way easier to have a basi Taiga up and running.
+
+For those of you who already have a self-hosted Taiga running with docker, this tutorial will guide you through the steps of the migration.
+
+[[taigadckr-to-taigadckr]]
+== Summary
+
+If you are familiar with docker and docker compose, this is the sensible summary:
+- save the modifications made in the `docker-compose.yml` and `docker-compose-inits.yml` files
```
Comment (daniel-herrero) [inline upgrades-docker-migrate.adoc:17]: I don't know if it deserves to warn about these files being overwritten, or if it'd be redundant.
Perhaps we can suggest to backup the files with other names

File: upgrades-docker-migrate.adoc [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+- Taiga secret key
+- Postgres connection
+- Email configuration
+- RabbitMQ configuration
+- Attachments' token expiration
+- Enable telemetry: we use this anonymous telemetry to improve Taiga. You may want to enable this to help us shape future Taiga.
+
+=== Add the customisation values to the `docker-file.yml` file
+
+Customisations are still in the `docker-compose.yml` file, for instance:
```
Comment (daniel-herrero) [inline upgrades-docker-migrate.adoc]: Perhaps we can add a link to the "Additional customization" section from here

File: upgrades-docker-migrate.adoc (hunk 10 above)
Comment (yamila-moreno) [reply upgrades-docker-migrate.adoc]: fixed!

File: upgrades-docker-migrate.adoc (hunk 11 above)
Comment (yamila-moreno) [reply upgrades-docker-migrate.adoc:21]: fixed!

File: upgrades-docker-migrate.adoc (hunk 13 above)
Comment (yamila-moreno) [reply upgrades-docker-migrate.adoc]: added it; maybe you can check if it's rendered ok?

Comment (daniel-herrero) [reply upgrades-docker-migrate.adoc]: Nice, it's working!

**Pull request merged**